### PR TITLE
Add support for `is_legacy` solo score index field

### PR DESCRIPTION
**Pull request by nanaya**
This also updates the BeatmapLeader to only consider lazer scores. User rank on score page (`Solo\Score#userRank`) will be updated separately.

- [x] production index needs to be recreated with updated indexer

### CODE REVIEW

File: app/Models/BeatmapLeader.php [hunk 1]
```diff
@@ -26,6 +26,7 @@ public static function sync(int $beatmapId, int $rulesetId): void
     {
         $searchParams = ScoreSearchParams::fromArray([
             'beatmap_ids' => [$beatmapId],
+            'is_legacy' => false,
```
Comment (notbakaneko) [inline app/Models/BeatmapLeader.php:29]: Do solo scores from stable have a `build_id`?

Comment (nanaya) [reply app/Models/BeatmapLeader.php:29]: no, it's what's being checked for is_legacy in the indexer itself as well.

Comment (notbakaneko) [reply app/Models/BeatmapLeader.php:29]: nvm, this is a different leader from what I was thinking of 🤨

**Review by notbakaneko — CHANGES_REQUESTED** [review]

**Review by notbakaneko — APPROVED** [review]
I was thinking could just make use of the `array_merge` already in `makeData` but this works.

**Comment by nanaya** [comment]
This has been approved but not merged 🤔

**Comment by notbakaneko** [comment]
is the index updated yet 👀

**Comment by nanaya** [comment]
it sure has since end of last month

**Comment by nanaya** [comment]
it means client is currently showing mixed scores for some rulesets until this is deployed 👀

**Pull request merged**